plugins/com.google.cloud.tools.eclipse.swtbot/src/com/google/cloud/tools/eclipse/swtbot/SwtBotTimeoutManager.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2011 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.swtbot;
+
+import org.eclipse.swtbot.swt.finder.utils.SWTBotPreferences;
+
+/**
+ * Used to set the general SWTBot timeout. This class was originally to be thrown away, except that
+ * SWTBot doesn't support a nice mechanism for temporarily changing the timeout.
+ * 
+ * Note that this is a "static" class, so it is not thread-safe. The pattern of usage is to call
+ * {@link #setTimeout(long)} to set the general SWTBot timeout, and then call
+ * {@link #resetTimeout()} to reset the SWTBot timeout back to it's original value. You should never
+ * call {@link #setTimeout(long)} more than once without a call to {@link #resetTimeout()} in
+ * between.
+ */
+public class SwtBotTimeoutManager {
+
+  private static final int TYPICAL_TIMEOUT = 30000;
+
+  private static long UNSET_TIMEOUT_VALUE = -1;
```

Comment:
We should probably treat any value <= 0 as an unset timeout, rather than requiring -1. What do you think?